client/src/components/ingredients/ReceiptField.tsx

```diff
@@ -0,0 +1,261 @@
+import {
+  Button,
+  Divider,
+  FormControl,
+  FormLabel,
+  Grid,
+  GridItem,
+  HStack,
+  Icon,
+  IconButton,
+  Image,
+  Text,
+} from '@chakra-ui/react';
+import { ChakraStylesConfig, Select } from 'chakra-react-select';
+import { MoveRight, Pencil, Trash } from 'lucide-react';
+import { Controller, FieldPath, Path } from 'react-hook-form';
+import { useDebouncedCallback } from 'use-debounce';
+import { ZodType, z } from 'zod';
+import CustomNumberInput from '../form/CustomNumberInput';
+import CustomTextInput from '../form/CustomTextInput';
+import { ReceiptFormProps } from './ReceiptForm';
+
+const menuStyles: ChakraStylesConfig = {
+  container: (baseStyles, state) => ({
+    ...baseStyles,
+    width: '100%',
+    border: 'none',
+  }),
+  control: (baseStyles, state) => ({
+    ...baseStyles,
+    border: 'none',
+  }),
+  indicatorsContainer: (baseStyles, state) => ({
+    backgroundColor: 'transparent',
+    background: 'transparent',
+  }),
+  indicatorSeparator: (baseStyles, state) => ({
+    backgroundColor: 'gray.600',
+  }),
+  dropdownIndicator: (baseStyles, state) => ({
+    backgroundColor: 'transparent',
+  }),
+  valueContainer: (baseStyles, state) => ({
+    ...baseStyles,
+    border: 'none',
+    paddingInline: '0',
+  }),
+};
+
+interface OptionType {
+  label: string;
+  value: any;
+}
+interface ReceiptFieldProps<T extends ZodType<any, any, any>> extends ReceiptFormProps<T> {
+  index: number;
+  field: z.infer<T>;
+}
+
+function ReceiptField<T extends ZodType<any, any, any>>({
+  control,
+  index,
+  remove,
+  update,
+  fields,
+  field,
+  watch,
+}: ReceiptFieldProps<T>) {
+  const updateReceipt = useDebouncedCallback((val: any) => {
+    console.log(val);
```

Comment:
Please make a mock API call to backend to help NA know. 
For example you could just write a line like 
const newData = await customFetch.get(something)
Do some update logic with the form to reflect it like modify the Options variable below or modify the field.suggested.item something.